pkg/glusterfs/controllerserver.go

```diff
@@ -29,6 +29,47 @@ type ControllerServer struct {
 	*GfDriver
 }
 
+// CsiDrvParam stores csi driver specific request parameters.
+// This struct will be used to gather specific fields of CSI driver:
+// For eg. csiDrvName, csiDrvVersion..etc
+// and also gather parameters passed from SC which not part of gluster volcreate api.
+// GlusterCluster - The resturl of gluster cluster
+// GlusterUser - The gluster username who got access to the APIs.
+// GlusterUserToken - The password/token of glusterUser to connect to glusterCluster
+// GlusterVersion - Says the version of the glustercluster running in glusterCluster endpoint.
+// CompMatrix - map which will be internally defined by the driver to make a compatibility
+//              version matrix between CSI driver and gluster cluster. All these fields are optional and can be used if needed.
+
+type CsiDrvParam struct {
+	GlusterCluster   string
+	GlusterUser      string
+	GlusterUserToken string
+	GlusterVersion   string
+	CsiDrvName       string
+	CsiDrvVersion    string
+	CompMatrix       map[string]string
+}
+
+// RequestConfig is the final struct after parsing request and CSI driver specific input
+type RequestConfig struct {
```

Comment:
Why not embed CsiDrvParam & api.VolCreateReq?

Reply:
api.VolCreateReq is the client API of GD2. So, embedding is not possible in it.